### DISCUSSION TITLE
Lie group normalization on `LieTensor`

### DISCUSSION THREAD

**Discussion by luisenp** [facebookresearch/theseus#507] [General]
Now that @fantaosha implemented normalization operator in #506, one question is what how do we want to use this in `LieTensor`. Some ideas:

1. All tensor inputs are checked for group properties. If broken, emit a warning and normalize.
2. All tensor inputs are normalized without checking. 
3. Constructor has an input flag that's something like `force_normalize`.
4. Constructor has an input flag that's something like `maybe_normalize` (check then normalize).
5. Add a context that global normalization behavior. Defaults to `True`. If `False` no checks/normalization. 

Something else? Looking for input from @mhmukadam @rmurai0610 @joeaortiz @dongjing3309 @maurimo @ddetone

**Comment by mhmukadam** [comment]
What tradeoff are we considering, for instance speed vs ease of use (too many warning, confusing api).

5 is a nice option for users to do this in chucks of code using `with`. Related to this we could consider a fully global context like `torch.set_default_tensor_type()`.

3/4 seems useful if users want to do this for any specific variable. For 4 `check_then_normalize` would be clearer.
